tests/integrationv2/test_client_authentication.py

```diff
@@ -11,116 +11,147 @@
+def test_client_auth_with_s2n_server_using_nonmatching_certs(managed_process, cipher, provider, protocol, certificate, client_certificate):
     port = next(available_ports)
 
+    if protocol < Protocols.TLS12 and client_certificate.algorithm == 'EC':
+        pytest.xfail("Client auth with ECDSA certs is current broken for versions < TLS1.2")
+
     client_options = ProviderOptions(
         mode=Provider.ClientMode,
         host="localhost",
         port=port,
         cipher=cipher,
-        curve=curve,
         data_to_send=b'',
         use_client_auth=True,
-        client_key_file=certificate.key,
-        client_certificate_file=certificate.cert,
+        key=client_certificate.key,
+        cert=client_certificate.cert,
+        trust_store=certificate.cert,
         insecure=False,
         protocol=protocol)
 
     server_options = copy.copy(client_options)
     server_options.data_to_send = None
     server_options.mode = Provider.ServerMode
-    server_options.key = Certificates.RSA_2048_SHA256_WILDCARD.key
-    server_options.cert = Certificates.RSA_2048_SHA256_WILDCARD.cert
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
 
     # Tell the server to expect the wrong certificate
-    server_options.client_certificate_file=Certificates.RSA_2048_SHA256_WILDCARD.cert
+    server_options.trust_store=Certificates.RSA_2048_SHA256_WILDCARD.cert
 
-    # Passing the type of client and server as a parameter will
-    # allow us to use a fixture to enumerate all possibilities.
     server = managed_process(S2N, server_options, timeout=5)
     client = managed_process(OpenSSL, client_options, timeout=5)
 
-    # OpenSSL should return 1 because the connection failed
+    # Openssl should tell us that a certificate was sent, but the handshake did not complete
     for results in client.get_results():
         assert results.exception is None
-        if protocol == Protocols.TLS13:
-            # Exit code 104 is connection reset by peer
-            # This is almost always 104, but we have hit an occasion where s_client
-            # closed cleanly.
-            assert results.exit_code == 104 or results.exit_code == 0
-        else:
-            assert results.exit_code == 1
-
-    # S2N should tell us that mutual authentication failed
+        assert b'write client certificate' in results.stderr
+        assert b'write certificate verify' in results.stderr
+        # TLS1.3 OpenSSL fails after the handshake, but pre-TLS1.3 fails during
+        if protocol is not Protocols.TLS13:
+            assert results.exit_code != 0
+            assert_openssl_handshake_complete(results, False)
+
+    # S2N should tell us that mutual authentication failed due to an untrusted cert
     for results in server.get_results():
         assert results.exception is None
-        assert results.exit_code == 255
+        assert results.exit_code != 0
```

Comment:
What are the different codes that are being thrown here?

Reply:
Are the possible exit codes and their meanings documented anywhere? I couldn't find any documentation, and generally it's only really safe to check if an return type is zero or nonzero.